File: app/src/main/java/com/penseapp/acaocontabilidade/chat/presenter/ChatsWriterPresenterImpl.java
package com.penseapp.acaocontabilidade.chat.presenter;

import com.penseapp.acaocontabilidade.chat.interactor.ChatsWriterInteractor;
import com.penseapp.acaocontabilidade.chat.interactor.ChatsWriterInteractorImpl;
import com.penseapp.acaocontabilidade.chat.view.ContactsView;

/**
 * Created by unity on 21/11/16.
 */

public class ChatsWriterPresenterImpl implements ChatsWriterPresenter {

    private final ContactsView contactsView;
    private final ChatsWriterInteractor chatsWriterInteractor;

    public ChatsWriterPresenterImpl(ContactsView contactsView) {
        this.contactsView = contactsView;
        this.chatsWriterInteractor = new ChatsWriterInteractorImpl(this);
    }

    @Override
    public void createChatIfNeeded(String senderId, String senderName, String senderCompany, String recipientId, String recipientName, String recipientCompany) {
        chatsWriterInteractor.createChatIfNeeded(senderId, senderName, senderCompany, recipientId, recipientName, recipientCompany);
    }

    @Override
    public void onChatCreated(String chatId, String chatName) {
        contactsView.onChatCreated(chatId, chatName);
    }
}

File: app/src/main/java/com/penseapp/acaocontabilidade/chat/chats/presenter/ChatsWriterPresenterImpl.kt
package com.penseapp.acaocontabilidade.chat.chats.presenter

import com.penseapp.acaocontabilidade.chat.chats.interactor.ChatsWriterInteractor
import com.penseapp.acaocontabilidade.chat.chats.interactor.ChatsWriterInteractorImpl
import com.penseapp.acaocontabilidade.chat.contacts.view.ContactsView

/**
 * Created by unity on 21/11/16.
 */

class ChatsWriterPresenterImpl(private val contactsView: ContactsView) : ChatsWriterPresenter {

    private val chatsWriterInteractor: ChatsWriterInteractor

    init {
        this.chatsWriterInteractor = ChatsWriterInteractorImpl(this)
    }

    override fun createChatIfNeeded(senderId: String,
                                    senderName: String,
                                    senderCompany: String,
                                    recipientId: String,
                                    recipientName: String,
                                    recipientCompany: String) {

        chatsWriterInteractor.createChatIfNeeded(senderId, senderName, senderCompany, recipientId,
                recipientName, recipientCompany)
    }

    override fun onChatCreated(chatId: String, chatName: String) {
        contactsView.onChatCreated(chatId, chatName)
    }
}
